.github/workflows/pr_test_peft_backend.yml

```diff
@@ -50,16 +45,19 @@ jobs:
       run: |
         apt-get update && apt-get install libsndfile1-dev libgl1 -y
         python -m pip install -e .[quality,test]
-        python -m pip install git+https://github.com/huggingface/accelerate.git
-        python -m pip install -U git+https://github.com/huggingface/transformers.git
-        python -m pip install -U git+https://github.com/huggingface/peft.git
+        if [ "${{ matrix.lib-version }}" == "main" ]; then
+            python -m pip install -U git+https://github.com/huggingface/peft.git
+            python -m pip install -U git+https://github.com/huggingface/transformers.git
+            python -m pip install -U git+https://github.com/huggingface/transformers.git
+        else
+            python -m pip install -U peft transformers accelerate
```

Comment:
Having a hard time understanding the thought process behind this. 


I'd consider this level of version discrepancy to be generally a red flag. Could you elaborate here a bit?

Reply:
this is just to make sure PRs we do in diffusers will not break anything for users that have the latest version of peft and transformers from pypi so it ensures backward compatbility. Right now we need to check these things manually which is a bit painful